pkg/parse/parse.go

```diff
@@ -1107,10 +1108,16 @@ func ContainerIgnoreFile(contextDir, path string, containerFiles []string) ([]st
 			return excludes, containerfileIgnore, err
 		}
 	}
-	path = filepath.Join(contextDir, ".containerignore")
+	path, symlinkErr := securejoin.SecureJoin(contextDir, ".containerignore")
+	if symlinkErr != nil {
+		return nil, "", symlinkErr
+	}
 	excludes, err := imagebuilder.ParseIgnore(path)
 	if errors.Is(err, os.ErrNotExist) {
-		path = filepath.Join(contextDir, ".dockerignore")
+		path, symlinkErr = securejoin.SecureJoin(contextDir, ".dockerignore")
```

Comment:
@nalind had objections to securejoin in this commit:  https://github.com/containers/buildah/pull/2627, I'm not sure if they still hold or what better alternatives there are now.  

I'll just go on the record noting that symlinks are utter evil!

Reply:
@TomSweeneyRedHat This is actually to prevent usage of symlink when its pointing outside the build-context ( so yes it assumes that symlink are bad and resolves it early and ignores if it points to a ignore file outside of build context )